### DISCUSSION TITLE
SABnzbd via SNAP: Howto

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2708] [General]
**SABnzbd via SNAP: Howto**

Snap is available for a lot of Linux distribution, like 

- Ubuntu
- Debian
- Raspberry Pi OS
- Fedora
- openSUSE
- Arch Linux
- Manjaro Linux

**Install snapd**

See https://snapcraft.io/docs/installing-snapd . On Debian and Raspi OS:

```
$ sudo apt update
$ sudo apt install snapd
```

**Install the sabznbd snap**

`$ sudo snap install sabnzbd`

**Run the sabnzbd snap**

`$ sudo snap run sabnzbd`

Access sabnzbd via http:127.0.0.1:8080

**Run the sabnzbd snap as normal user**

One time:
`sudo chmod 777 /var/snap/sabnzbd/common/sabnzbd.ini`

Then 

`$ snap run sabnzbd`

Access sabnzbd via http:127.0.0.1:8080

**Check if sabnzbd snap is running**

```
$ snap services | grep -i active | grep -i sabnzbd
sabnzbd.sabnzbd                                      enabled  active   -
```

**To do**

* run as normal user
* specify sabnzbd.ini
* set nice download dir
* autostart sabnzbd snap

**Comment by sanderjo** [comment]
On Raspi, if you get

`ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.`

do this:


sudo nano /etc/ld.so.preload

and commented the offending line out from

/usr/lib/arm-linux-gnueabihf/libarmmem.so

to

#/usr/lib/arm-linux-gnueabihf/libarmmem.so

**Reply by thezoggy** [reply]
could you not just run `sudo ldconfig`

**Reply by sanderjo** [reply]
please explain ...

**Reply by thezoggy** [reply]
if you are changing out linked libraries normally you would use ldconfig to clear out cache and relink stuff
https://man7.org/linux/man-pages/man8/ldconfig.8.html

so instead of manually updating to comment it out, what if you just run ldconfig to make it clear its cache?

**Reply by sanderjo** [reply]
I'll try as soon as I'm near that Raspi.

**Reply by sanderjo** [reply]
I tried:
un-uncommented the line to make it active, did **not** do the ldconfig command, rebooted raspi, started sabnzbd-snap ... no error at all. Weird.

**Reply by sanderjo** [reply]
Ah, it was back. So I tried `sudo ldconfig`, and did not help

```
sander@raspberrypi:~ $ snap run sabnzbd
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.

Fatal error:
  "/var/snap/sabnzbd/common/sabnzbd.ini" is not a valid configuration file<br>Error message: [Errno 13] Permission denied: '/var/snap/sabnzbd/common/sabnzbd.ini'
Specify a correct file or delete this file.
sander@raspberrypi:~ $ sudo ldconfig
sander@raspberrypi:~ $ snap run sabnzbd
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.

Fatal error:
  "/var/snap/sabnzbd/common/sabnzbd.ini" is not a valid configuration file<br>Error message: [Errno 13] Permission denied: '/var/snap/sabnzbd/common/sabnzbd.ini'
Specify a correct file or delete this file.
sander@raspberrypi:~ $ 


```

**Reply by thezoggy** [reply]
is that just bad platform detection in the snap make / unsupported platform?

guessing you found: https://stackoverflow.com/questions/42443273/raspberry-pi-libarmmem-so-cannot-open-shared-object-file-error/50958615#50958615 

found this, 
https://forum.snapcraft.io/t/snapcraft-3-x-cross-compilation-amd64-armhf-in-multipass-vm/9758/7

wonder if we need to add the run-on to fixup snap building for the correct arc instead of just using vm original arc

**Comment by sanderjo** [comment]
Running the sabnzbd snap as normal user:
```
$ snap run sabnzbd

Fatal error:
  "/var/snap/sabnzbd/common/sabnzbd.ini" is not a valid configuration file<br>Error message: [Errno 13] Permission denied: '/var/snap/sabnzbd/common/sabnzbd.ini'
Specify a correct file or delete this file.
```

solved / prevented with:

```
sander@zwart2204:~$ sudo chmod 777 /var/snap/sabnzbd/common/sabnzbd.ini

sander@zwart2204:~$ ll /var/snap/sabnzbd/common/sabnzbd.ini
-rwxrwxrwx 1 777 root 6171 okt 13 20:12 /var/snap/sabnzbd/common/sabnzbd.ini*

```

**Comment by sanderjo** [comment]
Something weird. Maybe @Safihre  and @jcfp  can help?

I can only start the sabnzbd snap via "snap run sabnzbd". Not too bad, but I can start the firefox snap and pycharm snap without "snap run".

```
sander@zwart2204:~$ which firefox
/snap/bin/firefox
```

... which is found as /snap/bin is in my path

Something weird: it's a link to the snap binary itself:

```
sander@zwart2204:~$ ll /snap/bin/firefox
lrwxrwxrwx 1 root root 13 okt 12 14:14 /snap/bin/firefox -> /usr/bin/snap*
```

@jcfp do you understand that?

Same with pycharm ... all good & automatic:

```
sander@zwart2204:~$ which pycharm-community 
/snap/bin/pycharm-community

sander@zwart2204:~$ ll /snap/bin/pycharm-community
lrwxrwxrwx 1 root root 13 okt 12 23:48 /snap/bin/pycharm-community -> /usr/bin/snap*
```


Back to sabnzbd: it's not in /snap/bin:

```
sander@zwart2204:~$ ll /snap/bin/| grep -i  sabnzbd
sander@zwart2204:~$ 
```

So ... is that the cause I cannot run it by typing just `sabnzbd` ?

**Reply by Safihre** [reply]
It should run on startup, so there's no command because it runs as daemon.

**Reply by sanderjo** [reply]
Ah! I did a reboot of Ubuntu, and indeed sabnzbd-snap is running. Nice!

```
$ snap services  | grep sabnzbd
sabnzbd.sabnzbd                                      enabled  active   -
```

**Comment by alfish2000** [comment]
Hi there!
Question to the snap:
Wouldn't it be possible somehow to use the snap "stable" and the "beta" channel for sabnzbd "Latest stable release" and "Latest pre-release"?
Just asking.
I'm quite new to snap etc but I have seen that other package are filling all snap channels like e.g. https://snapcraft.io/vlc

**Reply by Safihre** [reply]
AI tool found an easy way, so I just added it 🏅
Won't change current releases, so stable is still 5.0.0Beta3.

**Reply by alfish2000** [reply]
Great. Thanks. That was quick.
If I understand the https://github.com/sabnzbd/sabnzbd/commit/c43b11bd7c1a4e71ce96c83672cc41e685126a66 commit correctly it will move all "clean" tags with "major.minor.patch" to "stable" channel and the others to "candidate" channel.
Might the "beta" channel fit a little better when the other tag names also contain "beta"?

**Reply by Safihre** [reply]
Sure. I'll use the beta channel.

**Reply by alfish2000** [reply]
Great. Thanks.
I just found that page which explains these snap channels:
https://oneuptime.com/blog/post/2026-03-02-how-to-understand-snap-channels-stable-candidate-beta-and-edge/view

**Reply by alfish2000** [reply]
Just saw that you have "alpha" too.
Not sure how "alpha", "beta" would fit best in the 4 snap channels.
But I think most import is that a stable release goes to "stable" channel and these "alpha/beta" releases in another channel.

**Reply by alfish2000** [reply]
Hi again!
I just saw the beta4 here https://snapcraft.io/sabnzbd now on stable and beta channel.
The github automation should only have published to beta.
I'm curious if you have manually published to the stable channel also somehow to keep both channels equal until a new real release will come?

**Reply by Safihre** [reply]
Yes I did it manually. Like I said, this will only apply after the next stable release. Because current stable channel already holds 5.0.0 Betas.
